### ISSUE TITLE
[2.0] Cloak format for IPv6 CGI-IRC users is wrong

### ISSUE THREAD

**Issue by GLolol** [inspircd/inspircd#1249]
On my network running InspIRCd 2.0.x, IPv6 connections without rDNS should be cloaked in the format ` hidden-54cj3p.q13e.4qj0.3c01.2600.IP`, but CGI-IRC users with this setup instead get cloaks of the format `hidden-2qfrja`.

**Comment by Adam-** [comment]
What IP is being cloaked? What cloak settings are you using? Get the /whois output showing this.

**Comment by GLolol** [comment]
Regular connection:

```
14:53 -- [GLolol] (gl@hidden-54cj3p.q13e.4qj0.3c01.2600.IP): You can't be serious.
14:53 -- [GLolol] is connecting from gl@2600:3c01:e000:12c::8 2600:3c01:e000:12c::8
14:53 -- [GLolol] millennium.overdrivenetworks.com (A relatively long period of time... - Fremont, California)
14:53 -- [GLolol] is using modes +iwx
14:53 -- [GLolol] idle: 00 hours 00 minutes 09 seconds, signon at: Sat, 12 Nov 2016 14:53:01
14:53 -- [GLolol] End of /WHOIS list.
```

[Webchat](https://webirc.overdrivenetworks.com) user (different IP since the original person didn't want theirs posted here):
```
15:01 -- [ovdWebUser265] (webchat@hidden-moll24): Webchat / The Lounge User
15:01 -- [ovdWebUser265] is connecting from webchat@2600:8807:8400:004e:75ae:1a6e:8471:1a6d 2600:8807:8400:4e:75ae:1a6e:8471:1a6d
15:01 -- [ovdWebUser265] #chat
15:01 -- [ovdWebUser265] spacebus.overdrivenetworks.com (Spacebus Webchat Server)
15:01 -- [ovdWebUser265] is using modes +iwx
15:01 -- [ovdWebUser265] is using a secure connection
15:01 -- [ovdWebUser265] idle: 00 hours 00 minutes 12 seconds, signon at: Sat, 12 Nov 2016 15:01:46
15:01 -- [ovdWebUser265] End of /WHOIS list.
```

Cloaking configuration:
```
# cloak: requires m_md5 and m_conn_umodes
<module name="m_cloaking.so">
<cloak mode="half"
       key="SY1aN<redacted>"
       prefix="hidden-">
```

**Comment by attilamolnar** [comment]
Fixed by @Adam- in #1266.
